tdrs-backend/tdpservice/users/migrations/0018_ofa_admin_permissions.py

```diff
@@ -0,0 +1,102 @@
+# Generated by Django 3.2.5 on 2021-08-16 14:10
+from django.contrib.auth.management import create_permissions
+from django.db import migrations
+
+from tdpservice.users.permissions import (
+    add_permissions_q,
+    delete_permissions_q,
+    get_permission_ids_for_model,
+    view_permissions_q
+)
+
+
+def create_perms(apps, schema_editor):
+    """Creates permissions for all installed apps.
+
+    This is needed because Django does not actually create any Content Types
+    or Permissions until a post_migrate signal is raised after the completion
+    of `manage.py migrate`. When this migration is run as part of a set for a
+    freshly created database, that signal will not run until all migrations are
+    complete - resulting in no permissions for any Group.
+
+    For more info: https://code.djangoproject.com/ticket/29843
+    """
```

Comment:
It is important to note this step. While we don't need to do this for any of the subsequent migrations - as we add new models we will likely need to do this for the particular model after its added via initial migration. At that same time we'll need to assign any model permissions to the existing groups.

Reply:
That link was very informative to follow and surprising to see updates on a 3 year old ticket. It fully explains the necessity for this function, awesome inclusion!